source/cpptooling/data/representation.d

```diff
@@ -1423,6 +1423,20 @@ const:
         this.setUniqueId(makeUniqueUSR);
     }
 
+    this(CppNs name, CppNsStack stack, CppClass[] classes, CFunction[] funcs, CppNamespace[] namespaces, CxGlobalVariable[] globals) {
+	    this.name_ = name;
+        this.stack = stack;
+        /*this.classes = classes;
+        this.funcs = funcs;
+        this.globals = globals;*/
+	    //this.stack = CppNsStack(stack.dup);
+    } 
+
+    CppNamespace dup() {
```

Comment:
This is probably a bug in the standard CppTestdouble plugin.

resideInNs() and equivalent functions does not work as expected, only returns top namespace (bar in foo::bar)